src/main/java/org/jenkinsci/plugins/codesonar/services/HubVersionService.java

```diff
@@ -0,0 +1,181 @@
+package org.jenkinsci.plugins.codesonar.services;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.Consts;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.ParseException;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.util.EntityUtils;
+import org.jenkinsci.plugins.codesonar.CodeSonarPublisher;
+import org.jenkinsci.plugins.codesonar.models.CodeSonarHubInfo;
+import org.jenkinsci.plugins.codesonar.models.VersionCompatibilityInfo;
+
+import com.google.common.base.Throwables;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+import hudson.AbortException;
+
+public class HubVersionService {
+    private static final Logger LOGGER = Logger.getLogger(HubVersionService.class.getName());
+    
+    private HttpService httpService;
+
+    public HubVersionService(HttpService httpService) {
+        this.httpService = httpService;
+    }
+
+    public CodeSonarHubInfo getHubVersion(URI baseHubUri) throws AbortException {
```

Comment:
I think you should rename this 'getHubVersion()` method to be `fetchHubInfo()` to reflect the fact that it is not an ordinary `getter` (it must access the network, and it might fail), and also that it returns some sort of hub information, and not some sort of version type.